### PR TITLE
Add Kubernetes/IKS support

### PR DESCRIPTION
**Pull request by bhpratt**
### Description

Main change is adding Helm SCC api capabilities check to prevent SCCs from being applied to a Kubernetes (as opposed to OpenShift target cluster): `{{- if .Capabilities.APIVersions.Has "security.openshift.io/v1/SecurityContextConstraints" }}`

The remaining changes are to add a test case for a Kubernetes cluster and to modify the examples to allow either OpenShift or Kubernetes cluster creation.

One particular piece I'd like to call attention to is the resource group. As written, I could not get the test case resource groups to succeed. So I changed the pr_test resource group to `default` and commented out the resource group text in the examples/basic main.tf. However, open to feedback on how to make this piece work.

Specifically, this block didn't work as written:

```  
resource_group_name          = var.resource_group == null ? "${var.prefix}-resource-group" : null
existing_resource_group_name = var.resource_group
```  

Error:

```
TestRunBasicAgentsKubernetes 2024-03-22T14:45:40-04:00 logger.go:66: data.ibm_container_cluster_versions.cluster_versions: Read complete after 0s [id=2024-03-22 18:45:40.159623 +0000 UTC]
TestRunBasicAgentsKubernetes 2024-03-22T14:45:40-04:00 logger.go:66: 
TestRunBasicAgentsKubernetes 2024-03-22T14:45:40-04:00 logger.go:66: Planning failed. Terraform encountered an error while generating this plan.
TestRunBasicAgentsKubernetes 2024-03-22T14:45:40-04:00 logger.go:66: 
TestRunBasicAgentsKubernetes 2024-03-22T14:45:40-04:00 logger.go:66: ╷
TestRunBasicAgentsKubernetes 2024-03-22T14:45:40-04:00 logger.go:66: │ Error: [ERROR] Given Resource Group is not found in the account : %!s(<nil>) {
```

Which makes sense, as this resource group in `pr_test.go` does not exist in my account: `"geretain-test-observability-agents"`

Local tests:

```
--- PASS: TestRunUpgrade (3112.37s)
PASS
ok  	github.com/terraform-ibm-modules/terraform-ibm-observability-agents	3113.620s
```

```
TestRunBasicAgents 2024-03-22T14:10:08-04:00 tests.go:433: END: Destroy
--- PASS: TestRunBasicAgents (2861.70s)
```

```
TestRunBasicAgentsKubernetes 2024-03-22T13:55:32-04:00 tests.go:433: END: Destroy
--- PASS: TestRunBasicAgentsKubernetes (1985.95s)
```

### Release required?

- [ ] No release
- [ ] Patch release (`x.x.X`)
- [ ] Minor release (`x.X.x`)
- [ ] Major release (`X.x.x`)

### Run the pipeline

If the CI pipeline doesn't run when you create the PR, the PR requires a user with GitHub collaborators access to run the pipeline.

Run the CI pipeline when the PR is ready for review and you expect tests to pass. Add a comment to the PR with the following text:

```
/run pipeline
```

### Checklist for reviewers

- [ ] If relevant, a test for the change is included or updated with this PR.
- [ ] If relevant, documentation for the change is included or updated with this PR.

### For mergers

- Use a conventional commit message to set the release level. Follow the [guidelines](https://terraform-ibm-modules.github.io/documentation/#/merging.md).
- Include information that users need to know about the PR in the commit message. The commit message becomes part of the GitHub release notes.
- Use the **Squash and merge** option.

### CODE REVIEW

**Comment by ocofaigh** [comment]
@bhpratt Firstly, thank you very much for the PR!

FYI, the tests are currently designed specifically to run against our account, so hence why you saw the issues with the existing resource group since this group exists in our account.
If you were to execute the example itself and not pass any value for the `resource_group`, it would actually create the resource group for you, so hence could run in any account.
We have an issue in our backlog to see if we could make the tests more flexible to run in any account - its just something we have not been able to revisit yet.

In terms of the PR content, I think it is probably fine, but have you confirmed that functionality is working fine for a vanilla kube cluster set up (logs and metrics flowing fine)? 

We should probably update the main README.md to explicitly now call our vanilla kube support - Is it something you would like to include in this PR?

Thanks

**Comment by bhpratt** [comment]
@ocofaigh 

Re: Tests - Yes, logging and metrics flowed as expected. The only functional difference for the terraform code is that this prevents SCCs from attempting (and failing) to be applied to non-OpenShift clusters.

re: resource group - thanks for the backstory. That makes sense.

I think the problem is the test case passes in `geretain-test-observability-agents` every time as the resource group. I can see why this works if the resource group is in the account, because this check: `resource_group_name          = var.resource_group == null ? "${var.prefix}-resource-group" : null` will go to null and it will fall down to this check: `existing_resource_group_name = var.resource_group`

That will succeed in the test account. But in my account that will also fail because my account doesn't have `geretain-test-observability-agents`

I spent some time trying to come up with a way of using both `resource_group_name` and `existing_resource_group_name` in the data source (with test cases running in parallel), but couldn't come up with anything. My suggestion would be to use default or to have each test case create its own resource group. 

And good suggestion on README. I'll update that and also clean up a couple of smaller things I've noticed since opening up the PR.

**Review by ocofaigh — CHANGES_REQUESTED** [review]
hey, sorry about delay I had some pending comment but never hit the button to post them. Please address latest comments

File: examples/basic/main.tf [hunk 1]
```diff
@@ -68,18 +69,20 @@ resource "ibm_resource_instance" "cos_instance" {
 # Lookup the current default kube version
 data "ibm_container_cluster_versions" "cluster_versions" {}
 locals {
-  default_ocp_version = "${data.ibm_container_cluster_versions.cluster_versions.default_openshift_version}_openshift"
+  # default_ocp_version = "${data.ibm_container_cluster_versions.cluster_versions.default_openshift_version}_openshift"
```
[the remark this answers is not in the document]
Comment (bhpratt) [reply examples/basic/main.tf]: done

**Comment by bhpratt** [comment]
@ocofaigh removed commented out code and restored originals.

**Review by ocofaigh — APPROVED** [review]

**Pull request merged**